AddonTerraformTemplate/main.tf

```diff
@@ -0,0 +1,448 @@
+# Generate random resource group name
+resource "random_pet" "rg_name" {
+  prefix = var.resource_group_name_prefix
+}
+
+resource "azurerm_resource_group" "rg" {
+  location = var.resource_group_location
+  name     = random_pet.rg_name.id
+}
+
+resource "azurerm_kubernetes_cluster" "k8s" {
+  location            = azurerm_resource_group.rg.location
+  name                = var.cluster_name
+  resource_group_name = azurerm_resource_group.rg.name
+
+
+  dns_prefix          = var.dns_prefix
+  tags                = {
+    Environment = "Development"
+  }
+
+  default_node_pool {
+    name       = "agentpool"
+    vm_size    = "Standard_D2_v2"
+    node_count = var.agent_count
+  }
+
+  monitor_metrics {
+    annotations_allowed = "string"
+    labels_allowed = "string"
```

Comment:
Thanks Soham, could we make the annotations_allowed and labels_allowed as parameters with the default value as an empty string, to match the ARM and CLI onboarding? Also do you have a link to the terraform documentation for this? Who adds the terraform support for the new azure monitor metrics profile?

Reply:
Thanks Grace! I parameterized the annotations_allowed and labels_allowed similar to the other templates, in the latest revision. Here is the [link](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#monitor_metrics) for the docs. These resources are added in terraform by the swagger files. The respective teams are responsible for keeping the swagger files updated in Terraform, it is AKS team for the new azure monitor metrics profile.